### PR TITLE
Agnes weather app 

### PR DESCRIPTION
**Pull request by aagneshansson**
Link to my deployed project:
https://agnesweatherapp.netlify.app/

My reflections on how this week's project turned out:
The most fun project so far during this bootcamp. 

Things I'd like to have clarified or explained in more detail:
Map and more examples how to use it in relation to other functions.

### CODE REVIEW

File: code/index.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        <li class="nav-item">
+          <a href="./lysekil.html" class="nav-link">LYSEKIL WEATHER</a>
+        </li>
+      </ul>
+    </nav>
+  </header>
+  <div class="weathertext">
+    <h1 class="welcomeMessage">YOUR ULTIMATE WEATHER GUIDE</h1>
+  </div>
+  </main>
```
Comment (petrame) [inline code/index.html:40]: There is no `<main>` starting tag.

File: code/lysekil.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  </section>
+  <section class="forecast" id="forecast">
+
+    </div>
+    <div class="forecastLysekil" id="forecastLysekil">
+      <div id="forecastIcon" class="forecastIcon"></div>
+    </div>
+
+    </div>
```
Comment (petrame) [inline code/lysekil.html:52]: I think there is two more `</div>` than it should be.

File: code/style.css [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+      margin: 35px;
+      border-bottom: 1px white 1.5;
+    }
+    .todaysWeather, .todaysLysekil {
+      font-size: 30px;
+      text-align: center;
+    }
+    .geodaytime, .daytime, .forecastWeather, .forecastLysekil {
+      font-size: 25px;
+    }
```
Comment (petrame) [inline code/style.css:225]: Nice structured css file. Comments and logic names makes it easy to read.

File: README.md [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
 
 ## The problem
 
 Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+As I found this as a very fun and challenging project, I decided to try to achieve black level. I used two different api's to fetch the weather for Lysekil, my home town. Additionally I used if/else functions to change both background and text depending on the weather and temperature. Instead of spending to much time on the design I choosed to instead spend time to expand to fetching weather forecast by geographic coordinates. I created similar values for the geographic coordinates as for Lysekils weather. Lastly I got inspirated by another page I saw, and I decided to create a nav bar (as an animation) and add multiple pages so that I could navigate through different site depending on what weather I'd like to display. I'd say that this was the most challenging part, and even though I spent more time on that part than on the weather api's I'd enjoyed it a lot and I think I learnt so much during this process.
```
Comment (petrame) [inline README.md:8]: I think it's great that you decided in advance what you would spend your time on. It's easy to get caught up in the styling. And your page still looks really nice :)

File: code/geo.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    menuNav.classList.toggle("show");
+    menuBranding.classList.toggle("show");
+    navItems.forEach(item => item.classList.toggle("show"))
+}
+
+menuBtn.addEventListener("click", toggleMenu);
+
+// The function returns a color depending on the temperature 
+coloringFunction = (temp) => {
+    if (temp > 30.0) {
```
Comment (petrame) [inline code/geo.js:16]: I believe it's good practice to declare functions with const even if it obviously works without.

File: code/geo.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    } else if (temp > 0.0) {
+        return "#A73872";
+
+    } else if (temp < 0.0) {
+        return "#950952";
+
+    } else {
+        return "#9F1F62";
+    }
+};
```
Comment (petrame) [inline code/geo.js:34]: Nice feature with different color depending on temperature.

File: code/script.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    menu.classList.toggle("show");
+    menuNav.classList.toggle("show");
+    menuBranding.classList.toggle("show");
+    navItems.forEach(item => item.classList.toggle("show"))
+}
+
+menuBtn.addEventListener("click", toggleMenu);
+
+// Open or close menu by clicking or touching
+menuBtn.addEventListener("click", toggleMenu);
```
Comment (petrame) [inline code/script.js:15]: Line 12 and 15 is duplicates.

File: code/style.css [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+}
+
+/* This shows the menu onclick */
+
+.menu .show {
+  transform: translate3d(0, 0, 0);
+}
+
+/* Items inside of the nav */
+
```
Comment (petrame) [inline code/style.css:139]: Great way of commenting!

File: code/style.css [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+}
+
+.close .btn-line::before {
+  transform: translateY(12px) rotate(135deg);
+}
+
+.close .btn-line::after {
+  transform: translateY(-12px) rotate(-135deg);
+}
+
```
Comment (petrame) [inline code/style.css:71]: Love the rotation animation and the fold out menu.

**Review by petrame — COMMENTED** [review]
Lovely weather app with some cool animations and styling, I really like the idea with a landing page. A fav-icon had been the icing on the cake :)

It was interesting to check out your code since you did some features that I didn't have time for. The way of doing the hamburger menu was new to me.

Your code is well structured with easy-to-understand-comments. I wonder if there is a way to avoid duplication of code when you have multiple htlm and javascript files, but that's maybe something for the real code coaches to comment on.

Remember to remove your console logs when you are done.

**Review by petrame — APPROVED** [review]
Forgot to change status of the PR.

File: code/geo.html [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+      <div class="menu-branding" id="menuBranding">
+        <div class="portrait"></div>
+      </div>
+      <ul class="menu-nav" id="menuNav">
+        <li class="nav-item">
+          <a href="./index.html" class="nav-link">HOME</a>
+        </li>
+        <li class="nav-item current">
+          <a href="./geo.html" nav-link>CURRENT POSITION</a>
+        </li>
```
[the remark this answers is not in the document]
Comment (EmOlsz) [reply code/geo.html:30]: Yep, probably some typo

File: code/geo.html [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+        <li class="nav-item">
+          <a href="./lysekil.html" class="nav-link">LYSEKILS WEATHER</a>
+        </li>
+      </ul>
+    </nav>
+  </header>
+
+
+  <section class="geo" id="geo">
+    <div class="todaysGeo" id="todaysGeo"></div>
```
Comment (EmOlsz) [inline code/geo.html:40]: classes should be written in `kebab-case`, not `camelCase`

File: code/geo.js [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+            let weather = (todaysgeo.weather[0].description);
+
+            // Declaring the icon
+            const icon = `<img src=https://openweathermap.org/img/wn/${todaysgeo.weather[0].icon}@2x.png></img>`;
+            // Assign the returned color to become background color
+            document.body.style.backgroundColor = coloringFunction(temp);
+
+
+            // Displays time for sunrise and sunset and changing it to english timings
+            let sunrise = (new Date(todaysgeo.sys.sunrise * 1000).toLocaleTimeString("en-US", { timeStyle: "short" }));
```
Comment (EmOlsz) [inline code/geo.js:65]: No need to wrap everything in parentheses.
Moreover, I can't see any place where you are overwriting value of `sunrise` variable, so it should be declared with `const`

File: code/geo.js [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+
+    // Geolocation forecast weather
+    fetch(apiGeoForecast).then((response) => {
+        return response.json()
+    })
+        .then((geoforecast) => {
+            console.log(geoforecast);
+            const filteredGeoForecast = geoforecast.list.filter(item => item.dt_txt.includes('12:00'))
+
+            console.log(filteredGeoForecast);
```
Comment (EmOlsz) [inline code/geo.js:81]: Delete all console.logs before pushing your code to GitHub

File: code/style.css [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+header {
+  position: fixed;
+  z-index: 2;
+  width: 100%;
+}
+
+.main {
+  background: url(background.jpg);
```
Comment (EmOlsz) [inline code/style.css:22]: It's far more safe to use quotes around url, some browsers might have problem with reading path without them :

```
background: url("background.jpg");
```

File: code/style.css [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+}
+
+.welcomeMessage {
+  color: white;
+  padding-top: 150px;
+  margin: 10px;
+  font-size: 40px;
+}
+
+#home {
```
Comment (EmOlsz) [inline code/style.css:93]: We shouldn't style by id's

File: code/style.css [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+  padding-top: 45px;
+  padding-bottom: 40px;
+  margin: 35px;
+  border-bottom: 1px solid white;
+  display: grid;
+  text-align: center;
+}
+
+.forecast, .geoforecast, .forecastGeo {
+  padding-top: 0px;
```
Comment (EmOlsz) [inline code/style.css:174]: `padding-top: 0;`

File: code/style.css [hunk 17]
```diff
@@ -0,0 +1,225 @@
...
+  }
+  .todaysWeather, .todaysLysekil {
+    font-size: 25px;
+    text-align: center;
+  }
+  .geodaytime, .daytime, .forecastWeather, .forecastLysekil {
+    font-size: 25px;
+  }
+  /* Desktop device */
+  @media (min-width: 1025px) {
```
Comment (EmOlsz) [inline code/style.css:207]: You started to nest one `@media` inside another because you forgot about closing bracket in line 206

**Review by EmOlsz — CHANGES_REQUESTED** [review]
It's pretty impressive code, I have to say that Agnes! A lot of extra requirements done :)
However, with such complexity you didn't avoid many mistakes. Your code is a bit chaotic and has a lot of small mistakes. Please, take a look on both Petra's and my comments and provide changes

**Comment by EmOlsz** [comment]
Thank you for your changes Agnes!
Still waiting for the last change https://github.com/Technigo/project-weather-app/pull/138#discussion_r507640037
After fixing this issue, I will approve your project!

File: code/geo.js (hunk 12 above)
Comment (aagneshansson) [reply code/geo.js:65]: Implemented 😄

**Review by EmOlsz — APPROVED** [review]

**Pull request closed without merging**